core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/DataSource.scala

```diff
@@ -30,6 +30,34 @@ class SlickDataSource(val databaseConfig: DatabaseConfig[JdbcProfile])(implicit
     database.run(f(dataAccess).transactionally.withTransactionIsolation(isolationLevel))
   }
 
+  def createTempTable = {
+    sql"""call createEntityAttributeTempTable()""".as[Boolean]
+  }
+
+  def dropTempTable = {
+    sql"""call dropEntityAttributeTempTable""".as[Boolean]
+  }
+
+  // creates the ENTITY_ATTRIBUTE_TEMP for use by this transaction, executes the transaction, drops the temp table
+  def inTransactionWithAttrTempTable[T](f: (DataAccess) => ReadWriteAction[T], isolationLevel: TransactionIsolation = TransactionIsolation.RepeatableRead): Future[T] = {
+
+    val callerAction = f(dataAccess).transactionally.withTransactionIsolation(isolationLevel)
+
+    val callerActionWithTempTables = (for {
+      _ <- dropTempTable
+      _ <- createTempTable
+      origResult <- callerAction
+    } yield {
+      origResult
+    }).andFinally(dropTempTable)
+
+    database.run(callerActionWithTempTables.withPinnedSession).recover {
```

Comment:
David had a TODO to possibly have more complex error handling here. I'm not really sure what to do for it other than log an error but I do agree that it feels a bit bare given the complexity

Reply:
we can start here - I'm not sure what else to do either, we can monitor it as it runs